### PR TITLE
Fix compilation for older gcc

### PR DESCRIPTION
**Pull request by aburmash**
Signed-off-by: Alex Burmashev <alexander.burmashev@oracle.com>

### CODE REVIEW

File: shim.c [hunk 1]
```diff
@@ -1096,7 +1096,7 @@ EFI_STATUS start_image(EFI_HANDLE image_handle, CHAR16 *ImagePath)
 	void *sourcebuffer = NULL;
 	UINT64 sourcesize = 0;
 	void *data = NULL;
-	int datasize;
+	int datasize = 0;
```
Comment (jsetje) [inline shim.c:1099]: Should this be a separate commit?

Comment (aburmash) [reply shim.c:1099]: This still falls under category "fix compilation for older gcc". This is an issue for 4.8.5 and older(?) gcc, new enough will be fine with just int datasize;

Comment (jsetje) [reply shim.c:1099]: Actually this is fine, it is always set before reading.

**Review by jsetje — APPROVED** [review]

**Comment by vathpela** [comment]
Pushed as 4033d1fd0f25196f9a35e944679676277cd51960, which fixes the one conflict in the OpenSSL Makefile, and adds one workaround for gcc < 9 not having __builtin_strnlen().
